### ISSUE TITLE
DB error when upgrading 3.0.0 Beta3 to 3.0.2, using SQLite

### ISSUE THREAD

**Issue by werkstrom** [thorsten/phpMyFAQ#1756]
In the third step of the upgrade process I run into an error:

phpMyFAQ warning [2]: SQLite3::query(): Unable to execute statement: column config_name is not unique in /home/jdriahxo/public_html/experiencenew/src/phpMyFAQ/Database/Sqlite3.php on line 211

phpMyFAQ warning [2]: SQLite3::query(): Unable to execute statement: column config_name is not unique in /home/jdriahxo/public_html/experiencenew/src/phpMyFAQ/Database/Sqlite3.php on line 211

Fatal error: Uncaught TypeError: Return value of phpMyFAQ\Configuration::delete() must be of the type bool, object returned in /home/jdriahxo/public_html/experiencenew/src/phpMyFAQ/Configuration.php:444 Stack trace: #0 /home/jdriahxo/public_html/experiencenew/setup/update.php(683): phpMyFAQ\Configuration->delete('socialnetworks....') #1 {main} thrown in /home/jdriahxo/public_html/experiencenew/src/phpMyFAQ/Configuration.php on line 444

**Comment by werkstrom** [comment]
![image](https://user-images.githubusercontent.com/13434693/82359499-410cdf80-9a08-11ea-99f4-a921ed9fa87d.png)
Appears the table exists with the correct columns properties allready... This is from an old backup.

**Comment by werkstrom** [comment]
![image](https://user-images.githubusercontent.com/13434693/82359780-aa8cee00-9a08-11ea-884b-9b972a9d8757.png)
Same goes for instances

**Comment by thorsten** [comment]
yeah, version 3.0.0 removes some entries in the faqconfig table. We used the same method during the 2.9 releases as well.

**Comment by thorsten** [comment]
This is the method:
```
public function delete($name): bool
    {
        $delete = sprintf(
            "DELETE FROM
                %s%s
            WHERE
              config_name = '%s'",
            Database::getTablePrefix(),
            $this->tableName,
            $this->getDb()->escape(trim($name))
        );

        return $this->getDb()->query($delete);
}
```

**Comment by werkstrom** [comment]
Any way to log the entire SQL statement? If that's the problem...

**Comment by werkstrom** [comment]
Well.. as always I'm kind of lost in this PHP code ... Sorry.

**Comment by werkstrom** [comment]
Does appear as if: socialnetworks.enableFacebookSupport is removed from the DB though

**Comment by werkstrom** [comment]
OK... Did a dirty workaround and got it up and running...
![image](https://user-images.githubusercontent.com/13434693/82369435-51788680-9a17-11ea-8691-079d2337fef9.png)

**Comment by thorsten** [comment]
@werkstrom could you please try this:

```
    public function delete($name): bool
    {
        $delete = sprintf(
            "DELETE FROM
                %s%s
            WHERE
              config_name = '%s'",
            Database::getTablePrefix(),
            $this->tableName,
            $this->getDb()->escape(trim($name))
        );

        return (bool)$this->getDb()->query($delete);
    }
```

Thank you!

**Comment by werkstrom** [comment]
Can I run the upgrade script even though the installation is already updated? As per above I managed to sneek past this error and upgrade my installation.

**Comment by werkstrom** [comment]
OK... There appears to be an issue with accessing the update page when update is not possible/needed... However, that's very minor 😊

So, I did download the 3.0.3 (as I was running 3.0.2) and then the update page was accessible again. I added the (bool) as per your suggestion and update now runs smoothly. Thanks a lot! 👍👌☀

![image](https://user-images.githubusercontent.com/13434693/82729388-64c97180-9cf7-11ea-883d-96d0f9939333.png)

**Comment by thorsten** [comment]
You can always run the update as it optimizes the database as well.

**Comment by werkstrom** [comment]
@thorsten Re running update also appear to be working in 3.0.3, just tested. (didn't for some reason in 3.0.2 in my installation)...

**Comment by thorsten** [comment]
Yes, as your tables and configurations are already updated.
